### DISCUSSION TITLE
.sync(retry interval)

### DISCUSSION THREAD

**Discussion by ChecksumDK** [mathieucarbou/MycilaNTP#6] [Announcements]
I cannot find any information about what the retry interval does. Is it retrying sync if connection to NTP server failed or is a recurrent update of time every eg. 12 hours?
Is the retry interval seconds or?

**Comment by mathieucarbou** [comment]
this is just an interval that is used to get the local time and once ntp data is received and installed, it calls an optional callback to tell the app that the time is synced.

default value is fine most of the case.

**Comment by ChecksumDK** [comment]
Alright. Thanks.
So I have to call .sync everytime I want to resync the time?
Will .isSynched reset everytime I do a .sync?

man. 9. dec. 2024, 16.43 skrev Mathieu Carbou ***@***.***>:

**Comment by mathieucarbou** [comment]
> Alright. Thanks. So I have to call .sync everytime I want to resync the time? Will .isSynched reset everytime I do a .sync? man. 9. dec. 2024, 16.43 skrev Mathieu Carbou ***@***.***>:
> […](#)
> this is just an interval that is used to get the local time and once ntp data is received and installed, it calls an optional callback to tell the app that the time is synced. default value is fine most of the case. — Reply to this email directly, view it on GitHub <[#5 (comment)](https://github.com/mathieucarbou/MycilaNTP/issues/5#issuecomment-2528437263)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AKIKNOPL523BSVB4LHUH2WT2EW3AVAVCNFSM6AAAAABTJDHUSGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKMRYGQZTOMRWGM> . You are receiving this because you authored the thread.Message ID: ***@***.***>

no, it won't. is is a one-time thing: it just installs the ntp servers and esp-idf sntp layer does the rest.

once you get a time anyway you should not need to resync.  you can just change the timezone if you need after.

Also, ESP32 on 80Mhz clock has a frequency deviation of less than ±10 ppm, so this is quite accurate for a long time.
